node/rpc/src/chain_api.rs

```diff
@@ -98,22 +102,32 @@ where
             )
             .await
     } else {
-        let file = File::from_std(tmp_file.reopen()?);
-        data.chain_store
-            .export(
-                &start_ts,
-                recent_roots,
-                AsyncWriterWithChecksum::<Sha256, _>::new(BufWriter::new(file)),
-            )
-            .await
+        let file = async_fs::File::from(tmp_file.reopen()?);
+        if compressed {
```

Comment:
Can we do it without duplicating this snippet by supplying an abstract writer?

Reply:
I meant something like this 
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=980b6908a0c22f335c6120a07ebb1374

Reply:
I c, but having a `Box` or `Arc` wrapper conflicts with the other change that makes `AsyncWriteWithChecksum` usage internal to the `export` function.

Reply:
Could u elaborate on ur suggestion [here](https://github.com/ChainSafe/forest/pull/2754#discussion_r1164161729)?

Reply:
Alright, if it's a massive pain then I'm good with the existing solution.

Reply:
I've reverted the other change and try to apply this one, but it's still hard to make it compile as `W` has 2 non-auto traits `AsyncWrite + Checksum<D>` 

Got `error[E0225]: only auto traits can be used as additional traits in a trait object` to compile the below code.  How about just keep the code as it was? 

```rust
let writer: Arc<TokioMutex<dyn AsyncWrite + Checksum<Sha256> + Send + Unpin>> =
            if compressed {
                Arc::new(TokioMutex::new(AsyncWriterWithChecksum::<Sha256, _>::new(
                    ZstdEncoder::new(BufWriter::new(file)),
                    false,
                )))
            } else {
                Arc::new(TokioMutex::new(AsyncWriterWithChecksum::<Sha256, _>::new(
                    BufWriter::new(file),
                    !skip_checksum,
                )))
            };
```